src/main/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImpl.java

```diff
@@ -1,36 +1,29 @@
 package net.syscon.elite.api.resource.impl;
 
-import net.syscon.elite.api.model.Agency;
 import net.syscon.elite.api.model.CourtHearing;
 import net.syscon.elite.api.model.PrisonToCourtHearing;
 import net.syscon.elite.api.resource.OffenderMovementsResource;
-import org.springframework.http.HttpStatus;
+import net.syscon.elite.service.CourtHearingsService;
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import static org.springframework.http.HttpStatus.CREATED;
+
 @RestController
 @RequestMapping("${api.base.path}/bookings")
 @Validated
 public class OffenderMovementsResourceImpl implements OffenderMovementsResource {
-    //
-    // TODO - WIP what role(s) do we need to enforce here?
-    //
+
+    private final CourtHearingsService courtHearingsService;
+
+    public OffenderMovementsResourceImpl(final CourtHearingsService courtHearingsService) {
+        this.courtHearingsService = courtHearingsService;
+    }
+
     @Override
     public ResponseEntity<CourtHearing> prisonToCourtHearing(final Long bookingId, final PrisonToCourtHearing hearing) {
-        //
-        // TODO - WIP this is currently a stubbed response
-        //
-        return ResponseEntity.status(HttpStatus.CREATED).body(CourtHearing.builder()
-                .id(-1L)
-                .date(hearing.getCourtHearingDateTime().toLocalDate())
-                .time(hearing.getCourtHearingDateTime().toLocalTime())
-                .location(Agency.builder()
-                        .agencyId(hearing.getToCourtLocation())
-                        .active(true)
-                        .agencyType("CRT")
-                        .build())
-                .build());
+        return ResponseEntity.status(CREATED).body(courtHearingsService.scheduleHearing(bookingId, hearing));
```

Comment:
Can you just return a CourtHearing? I guess there is an annotation to say 201 response code?

Reply:
The challenge in (some respects) is looking for consistency in the elite code base.  It isn't always consistent.